src/mscorlib/src/System/MarshalByRefObject.cs

```diff
@@ -256,7 +244,28 @@ internal static bool CanCastToXmlTypeHelper(RuntimeType castType, MarshalByRefOb
 
             return o.CanCastToXmlType(xmlTypeName, xmlTypeNamespace);
         } // CanCastToXmlType
-
+#else
+        protected MarshalByRefObject() { }
```

Comment:
Thanks for pointing out the discussion (I'm way far behind on my github notifications), I just added a comment there that we should keep this member and it's closure in System.Runtime and thus implement them in System.Private.CoreLib.